server/prompt.go

```diff
@@ -17,7 +17,7 @@ import (
 func chatPrompt(ctx context.Context, r *runnerRef, msgs []api.Message) (prompt string, images []llm.ImageData, _ error) {
 	// pull out any system messages which should always be included in the prompt
 	var system []api.Message
-	msgs = slices.DeleteFunc(msgs, func(m api.Message) bool {
+	msgs = slices.DeleteFunc(append(r.model.Messages, msgs...), func(m api.Message) bool {
```

Comment:
This seems like over-coupling the `runnerRef` which is a scheduling thing to what needs to be a simple prompt truncation function. Could we do this before passing it in to `chatPrompt`?